x/pairing/keeper/grpc_query_provider_epoch_cu.go

```diff
@@ -0,0 +1,44 @@
+package keeper
+
+import (
+	"context"
+	"sort"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/x/pairing/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) ProviderEpochCu(goCtx context.Context, req *types.QueryProviderEpochCuRequest) (*types.QueryProviderEpochCuResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	infoMap := map[string]uint64{}
+
+	_, keys, pecs := k.GetAllProviderEpochCuStore(ctx)
+	for i := range pecs {
+		provider, _, err := types.DecodeProviderEpochCuKey(keys[i])
+		if err != nil {
+			continue
+		}
+		if _, ok := infoMap[provider]; !ok {
+			infoMap[provider] = pecs[i].ServicedCu
+		} else {
+			infoMap[provider] += pecs[i].ServicedCu
+		}
+	}
+
+	info := []types.ProviderCuInfo{}
+	for provider, cu := range infoMap {
```

Comment:
iteration on a map

Reply:
but I'm sorting it right after so it's ok